dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+
+            put("MM/dd/yyyy HH:mm",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2})\\\"?");
+
+            put(datetimeFormat.getPattern(),
+                    "\\\"?(\\d{4}\\d{2}\\d{2}\\d{2}\\d{2}\\d{2})\\\"?");
+
+            put(dateFormat.getPattern(), "\\\"?(\\d{4}\\d{2}\\d{2})\\\"?");
+
+            put("MM/dd/yyyy", "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4})\\\"?");
+
+            put("hh:mm:ssa", "\\\"?(\\d{1,2}:\\d{1,2}:\\d{1,2}(?:AM|PM|am|pm))\\\"?");
+
+            put("hh:mm:ss a", "\\\"?(\\d{1,2}:\\d{1,2}:\\d{1,2}\\s+(?:AM|PM|am|pm))\\\"?");
+
+            put("HH:mm:ss", "\\\"?(\\d{1,2}:\\d{1,2}:\\d{1,2})\\\"?");
+
+            put("hh:mma", "\\\"?(\\d{1,2}:\\d{1,2}(?:AM|PM|am|pm))\\\"?");
+
+            put("hh:mm a", "\\\"?(\\d{1,2}:\\d{1,2}\\s+(?:AM|PM|am|pm))\\\"?");
+
+            put("HH:mm", "\\\"?(\\d{1,2}:\\d{1,2})\\\"?");
+        }
+    };
+
+    private static final List<String> LUCENE_TIME_FORMAT_PATTERNS = new ArrayList<>(
+            Arrays.asList("hh:mm:ssa", "hh:mm:ss a", "HH:mm:ss", "hh:mma", "hh:mm a", "HH:mm"));
+    /**
+     *
+     * @param query
+     * @return
+     */
+    static String findAndReplaceQueryDates(String query) {
+        query = RegEX.replaceAll(query, " ", "\\s{2,}");
+
+        //delete additional blank spaces on date range
+        if(UtilMethods.contains(query, "[ ")) {
+            query = query.replace("[ ", "[");
+        }
+
+        if(UtilMethods.contains(query, " ]")) {
+            query = query.replace(" ]", "]");
+        }
+
+        String clausesStr = RegEX.replaceAll(query, "", "[\\+\\-\\(\\)]*");
+        String[] tokens = clausesStr.split(" ");
+        String token;
+        List<String> clauses = new ArrayList<String>();
+        for (int pos = 0; pos < tokens.length; ++pos) {
+            token = tokens[pos];
+            if (token.matches("\\S+\\.\\S+:\\S*")) {
+                clauses.add(token);
+            } else if (token.matches("\\d+:\\S*")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else if (token.matches("\\[\\S*")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + token);
+            } else if (token.matches("TO") || token.toLowerCase().matches("am") || token.toLowerCase().matches("pm")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else if (token.matches("\\S*\\]")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else if (token.matches("\\d{1,2}/\\d{1,2}/\\d{4}")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else {
+                clauses.add(token);
+            }
+        }
+
+        //DOTCMS - 4127
+        List<Field> dateFields = new ArrayList<Field>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'dateFields' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)